### DISCUSSION TITLE
Getting multiple recordsets from a stored procedure

### DISCUSSION THREAD

**Discussion by Kokainym** [SOCI/soci#1109] [Q&A]
Is it possible to access other recordsets or create vector of rowset from the result of executing a stored procedure?  

`rowset<row> rs = sql->prepare << "EXEC GetPatterns";`  
GetPatterns procedure return two recordsets with data. By this code I can access only to first recordset, but it returns a second or more.  

Thx :)

**Comment by vadz** [comment]
You need to iterate over the statement output in the usual way, AFAICS there is nothing special about this being a stored procedure or recordset. See "Statements" section of the docs.

**Comment by Kokainym** [comment]
Hi  
I tried to understand and figure it out for several days, but apparently my knowledge is still not enough :(( 
I would see it like this, but this code does not work. An analogy with Sequelize.


```
std::vector<rowset<row>> recordsets;

statement st = (sql.prepare << "EXEC GetPatterns 347", into(recordsets));
st.execute();

for (auto& rs : recordsets) {
	for (auto& row : rs) {
		std::cout << "id - " << row.get<int>(0) << ", data - " << row.get<std::string>(9) << std::endl;
	}
}
```
